src/core/types.h

```diff
@@ -17,6 +17,21 @@ enum class MemoryAccessSize : u32
   Word
 };
 
+struct DebugAddress
```

Comment:
Rather than combining the breakpoints, it'd be better to store the execution and data breakpoints in separate arrays (fewer loops/comparisons for every guest instruction).